tokens/usecase/tokens_usecase.go

```diff
@@ -41,7 +39,16 @@ type tokensUseCase struct {
 	pricingStrategyMap map[domain.PricingSourceType]domain.PricingSource
 
 	// Map of chain denoms to coingecko IDs
-	coingeckoIds map[string]string
+	coingeckoIds sync.Map // map[string]string
+
+	// Represents the interval at which to update the assets from the chain registry
+	updateAssetsHeightInterval int
+
+	// TokenRegistryLoader fetches tokens from the chain registry into the tokens use case
+	tokenLoader TokenRegistryLoader
+
+	// Logger instance
+	logger log.Logger
```

Comment:
Logger is no longer used at this moment, but I think it's still useful to have it. Please let me know if I should remove it.